app/controllers/responses_controller.rb

```diff
@@ -1,33 +1,44 @@
 class ResponsesController < ApplicationController
+  rescue_from BaseModel::RecordInvalid, with: :unprocessable_entity
+  rescue_from ActionController::ParameterMissing, with: :parameter_missing
+
   wrap_parameters Response, format: :json
 
   def create
     @response = Response.new(convert_response_params)
     render json: ResponseSerializer.new(@response).as_json
-  rescue ActionController::ParameterMissing
-    head :unprocessable_entity
   end
 
   private
 
   # TODO: update model to be equal to the model used by clientside so this conversion is unneccesary
   def convert_response_params
     {
-      answer_values: response_params['questions'].each_with_object({}) do |question, memo|
-        memo[question['key']] = question['answer_value']
-      end
+      answer_values: response_params[:questions].each_with_object({}) do |question, memo|
+        memo[question[:id]] = question[:answer_value]
+      end,
+      domain_ids: response_params[:domain_ids]
     }
   end
 
+  def parameter_missing(exception)
+    render json: { message: exception.message }, status: :unprocessable_entity
+  end
+
+  def unprocessable_entity(exception)
+    render json: { message: exception.message }, status: :unprocessable_entity
+  end
+
   def response_params
```

Comment:
Met strong parameters kun je attributes required maken, misschien kun je daar wat mee in het geval van `domain_ids`. Met `fetch` kun je ook optionele attributen een default value geven.

Om params van het formaat `{ questions: [{ 'id' => 'EL02', 'answer_value' => 1 }], domain_ids: ['POS-PQ'] }` om te zetten naar `{"EL02"=>1}` kun je dan gebruiken: `Hash[params[:questions].map{|q|q.values}]`.